Calculate CSS unit `x` as resolution for calc combination

CYBAI opened the pull request:
#### a4c5bcd2ebf9effc1d52f8dbc298f46b08c1eb64
<pre>
Calculate CSS unit `x` as resolution for calc combination
<a href="https://bugs.webkit.org/show_bug.cgi?id=257717">https://bugs.webkit.org/show_bug.cgi?id=257717</a>

Reviewed by Tim Nguyen.

Previously, `x` was calculated as `other`category so it cannot
be canonicalized to `dppx` unit correctly. Thus, in this patch,
we&apos;ll calculate `x` as `resolution` category in calc combination
to fix the issue.

* LayoutTests/imported/w3c/web-platform-tests/css/css-images/image-set/image-set-parsing-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-images/image-set/image-set-parsing.html:
* Source/WebCore/css/calc/CSSCalcCategoryMapping.cpp:
(WebCore::calculationCategoryForCombination):

Canonical link: <a href="https://commits.webkit.org/265469@main">https://commits.webkit.org/265469@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/7090fd773b7abeca2b8b7014dcaa61ee7491c5db

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/10955 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/11168 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/11486 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/12602 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/10464 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/10971 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/13545 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/11150 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/13390 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/11114 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/12013 "Passed tests") | [❌ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/9233 "2 api tests failed or timed out") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/13008 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/9306 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/9896 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/17126 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/10377 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/10049 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/13290 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/10490 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/8581 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/9664 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/9787 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/2635 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/13931 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/10346 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

nt1m reviewed (approved):
Thanks!

nt1m commented on LayoutTests/imported/w3c/web-platform-tests/css/css-images/image-set/image-set-parsing-expected.txt:
```diff
@@ -77,10 +77,10 @@ PASS e.style['background-image'] = "image-set(url(\"example.png\") 0dpi)" should
+FAIL e.style['background-image'] = "image-set(url(\"example.png\") calc(-1 * 1x))" should set the property value assert_equals: serialization should be canonical expected "image-set(url(\"example.png\") calc(-1 * 1x))" but got "image-set(url(\"example.png\") calc(-1dppx))"
+FAIL e.style['background-image'] = "-webkit-image-set(url(\"example.png\") calc(-1 * 1x))" should set the property value assert_equals: serialization should be canonical expected "image-set(url(\"example.png\") calc(-1 * 1x))" but got "image-set(url(\"example.png\") calc(-1dppx))"
+FAIL e.style['background-image'] = "image-set(url(\"example.png\") calc(1x + -1x))" should set the property value assert_equals: serialization should be canonical expected "image-set(url(\"example.png\") calc(1x + -1x))" but got "image-set(url(\"example.png\") calc(0dppx))"
+FAIL e.style['background-image'] = "-webkit-image-set(url(\"example.png\") calc(1x + -1x))" should set the property value assert_equals: serialization should be canonical expected "image-set(url(\"example.png\") calc(1x + -1x))" but got "image-set(url(\"example.png\") calc(0dppx))"
```
These should now pass if you want to resync that specific test from upstream: https://github.com/web-platform-tests/wpt/blob/ce1b8fa687/css/css-images/image-set/image-set-parsing.html

CYBAI replied:
ah, I was trying to figure out why it failed and didn't realize it needs a sync as well. thanks for catching it! I will handle it in this PR. 🙏  (for `css/mediaqueries`, I will handle it in https://bugs.webkit.org/show_bug.cgi?id=258438)

CYBAI replied:
Imported the tests in the new commit.

nt1m commented on LayoutTests/imported/w3c/resources/import-expectations.json:
```diff
@@ -128,6 +128,7 @@
     "web-platform-tests/css/css-grid/parsing/": "import",
     "web-platform-tests/css/css-highlight-api": "import",
     "web-platform-tests/css/css-images": "import",
+    "web-platform-tests/css/css-images/image-set/": "import",
```
Please undo this. It’s redundant with the previous line

CYBAI replied:
I just undo the line though, I'm wondering when the import expectations should be added, I noticed some other places might also have child folders in the import-expectations.json like following

https://github.com/WebKit/WebKit/blob/b50c5911015252922d4511ec940e88b1dba20354/LayoutTests/imported/w3c/resources/import-expectations.json#L155-L158

nt1m replied:
yeah, it's an unintended side effect of the script, ideally we should clean up the others as well

CYBAI committed:
Calculate CSS unit `x` as resolution for calc combination
https://bugs.webkit.org/show_bug.cgi?id=257717

Reviewed by Tim Nguyen.

Previously, `x` was calculated as `other`category so it cannot
be canonicalized to `dppx` unit correctly. Thus, in this patch,
we'll calculate `x` as `resolution` category in calc combination
to fix the issue.

* LayoutTests/imported/w3c/web-platform-tests/css/css-images/image-set/image-set-parsing-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-images/image-set/image-set-parsing.html:
* Source/WebCore/css/calc/CSSCalcCategoryMapping.cpp:
(WebCore::calculationCategoryForCombination):

Canonical link: https://commits.webkit.org/265469@main
